### PR TITLE
[Enhancement] (nereids)implement DropRoleCommand in nereids

### LINKED ISSUE #42618: [Enhancement] (nereids)implement DropRoleCommand in nereids
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

DropRoleCommand is supported in the legacy planner but needs implementation in Nereids. The syntax is defined in DorisParser.g4. For reference, examine DropRoleStmt.java in the legacy planner.

```unsupportedDropStatement : DROP ROLE (IF EXISTS)? name=identifier```

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Vallishp**
### What problem does this PR solve?

support drop role command in nereids

Issue Number: close #42618 

Related PR: #

Problem Summary:

### Check List (For Committer)

- Test 

    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 

test_grant_priv.groovy already cover it.
2024-11-04 15:47:00.905 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: drop role if exists test_grant_priv_role1
2024-11-04 15:47:00.908 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: DROP DATABASE IF EXISTS test_grant_priv_db
2024-11-04 15:47:00.913 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: CREATE DATABASE test_grant_priv_db
2024-11-04 15:47:00.921 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: CREATE ROLE test_grant_priv_role1
2024-11-04 15:47:00.928 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: CREATE USER 'test_grant_priv_user1' IDENTIFIED BY '123456'
2024-11-04 15:47:00.938 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: CREATE USER 'test_grant_priv_user2' IDENTIFIED BY '123456'
2024-11-04 15:47:00.946 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: grant select_priv on test_grant_priv_db.* to test_grant_priv_user1
2024-11-04 15:47:00.969 INFO [suite-thread-1] (SuiteContext.groovy:299) - Create new connection for user 'test_grant_priv_user1' to 'jdbc:mysql://127.0.0.1:9030/test_grant_priv_db?'
2024-11-04 15:47:01.016 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: grant select_priv on test_grant_priv_db.* to test_grant_priv_user2
2024-11-04 15:47:01.034 INFO [suite-thread-1] (test_grant_priv.groovy:46) - errCode = 2, detailMessage = Access denied; you need all  [Grant_priv, Select_priv] privilege(s) for this operation
2024-11-04 15:47:01.034 INFO [suite-thread-1] (SuiteContext.groovy:309) - Recover original connection to 'jdbc:mysql://127.0.0.1:9030/test_grant_priv_db?'
2024-11-04 15:47:01.035 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: grant grant_priv on test_grant_priv_db.* to test_grant_priv_user1
2024-11-04 15:47:01.042 INFO [suite-thread-1] (SuiteContext.groovy:299) - Create new connection for user 'test_grant_priv_user1' to 'jdbc:mysql://127.0.0.1:9030/test_grant_priv_db?'
2024-11-04 15:47:01.079 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: grant select_priv on test_grant_priv_db.* to test_grant_priv_user2
2024-11-04 15:47:01.090 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: grant load_priv on test_grant_priv_db.* to test_grant_priv_user2
2024-11-04 15:47:01.099 INFO [suite-thread-1] (test_grant_priv.groovy:63) - errCode = 2, detailMessage = Access denied; you need all  [Grant_priv, Load_priv] privilege(s) for this operation
2024-11-04 15:47:01.100 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: grant 'test_grant_priv_role1' to test_grant_priv_user2
2024-11-04 15:47:01.116 INFO [suite-thread-1] (test_grant_priv.groovy:70) - errCode = 2, detailMessage = Access denied; you need (at least one of) the (GRANT/REVOKE) privilege(s) for this operation
2024-11-04 15:47:01.116 INFO [suite-thread-1] (SuiteContext.groovy:309) - Recover original connection to 'jdbc:mysql://127.0.0.1:9030/test_grant_priv_db?'
2024-11-04 15:47:01.117 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: grant grant_priv on *.* to test_grant_priv_user1
2024-11-04 15:47:01.127 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: grant 'test_grant_priv_role1' to test_grant_priv_user2
2024-11-04 15:47:01.139 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: drop user if exists test_grant_priv_user1
2024-11-04 15:47:01.149 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: drop user if exists test_grant_priv_user2
2024-11-04 15:47:01.165 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: drop role if exists test_grant_priv_role1
2024-11-04 15:47:01.172 INFO [suite-thread-1] (Suite.groovy:384) - Execute sql: DROP DATABASE IF EXISTS test_grant_priv_db
2024-11-04 15:47:01.184 INFO [suite-thread-1] (ScriptContext.groovy:120) - Run test_grant_priv in /root/doris/workspace/doris/regression-test/suites/account_p0/test_grant_priv.groovy succeed
2024-11-04 15:47:09.541 INFO [main] (RegressionTest.groovy:281) - Start to run single scripts
2024-11-04 15:48:21.903 INFO [main] (RegressionTest.groovy:410) - Success suites:
/root/doris/workspace/doris/regression-test/suites/account_p0/test_grant_priv.groovy: group=default,p0, name=test_grant_priv

- Behavior changed:

    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?

    - [ ] No.
    - [ ] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/DropRoleCommand.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ */
+public class DropRoleCommand extends DropCommand {
+    public static final Logger LOG = LogManager.getLogger(DropRoleCommand.class);
+    private final boolean ifExists;
+    private final String role;
+
+    /**
+     * constructor
+     */
+
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/DropRoleCommand.java:46]: remove empty line
```suggestion
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/DropRoleCommand.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    public DropRoleCommand(String role, boolean ifExists) {
+        super(PlanType.DROP_ROLE_COMMAND);
+        this.role = role;
+        this.ifExists = ifExists;
+    }
+
+    @Override
+    public void doRun(ConnectContext ctx, StmtExecutor executor) throws Exception {
+
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/DropRoleCommand.java:55]: remove empty line
```suggestion
```

**Comment by Vallishp** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]
LGTM

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**